pacman-nodejs-app/docs/pacman-nodejs-app-aws-migration.md

```diff
@@ -0,0 +1,532 @@
+kubectl config use-context gke_${GCP_PROJECT}_us-west1-b_gce-us-west1
+```
+
+### Create and Use the pacman Namespace
+
+Create the namespace needed for the migration:
+
+```bash
+kubectl create -f pacman-ns-dump/ns.json
+```
+
+Set the namespace of the context:
+
+```bash
+kubectl config set-context gke_${GCP_PROJECT}_us-west1-b_gce-us-west1 --namespace pacman
+```
+
+### Create Pac-Man Kubernetes Resources
+
+Add the exported resources into the namespace:
+
+```bash
+kubectl create -f pacman-ns-dump/pacman-ns-dump.json
+```
+
+### Verify All Resources
+
+All resources are ready and available after all pods are in the `RUNNING` state or their corresponding deployments show the correct number of pods `AVAILABLE`.
+Also when all services show an `EXTERNAL-IP`.
+
+```bash
+kubectl get all -o wide
+```
+
+### Save Pac-Man and MongoDB Load Balancer IP
+
+```bash
+PACMAN_DST_PUBLIC_IP=$(kubectl get svc pacman --output jsonpath="{.status.loadBalancer.ingress[0].ip}")
+MONGO_DST_PUBLIC_IP=$(kubectl get svc mongo --output jsonpath="{.status.loadBalancer.ingress[0].ip}")
+```
+
+### Add New Mongo Instance to Replica Set
+
+Connect to the mongo pod in cluster A and invoke the mongo client to connect to the mongodb PRIMARY in order to add the new mongo instance as a replica set SECONDARY.
+
+```bash
+kubectl --context=subdomain.example.com \
+    exec -it ${MONGO_SRC_POD} -- \
+    mongo --eval "rs.add(\"${MONGO_DST_PUBLIC_IP}:27017\")"
+```
+
+### Check Status of New Mongo Instance In Replica Set
+
+```bash
+kubectl --context=subdomain.example.com  \
+    exec -it ${MONGO_SRC_POD} -- \
+    mongo --eval "rs.status()"
+```
+
+### Make New Mongo Instance Primary
+
+```bash
+kubectl --context=subdomain.example.com \
+    exec -it ${MONGO_SRC_POD} -- \
+    mongo --eval "rs.stepDown(120)"
+```
+
+### Update DNS
+
+Update DNS to point to new cluster:
+
+```bash
+gcloud dns record-sets transaction start -z=${ZONE_NAME}
+gcloud dns record-sets transaction remove -z=${ZONE_NAME} --name="pacman.${DNS_NAME}" --type=CNAME --ttl=1 "${PACMAN_SRC_PUBLIC_IP}"
+gcloud dns record-sets transaction add -z=${ZONE_NAME} --name="pacman.${DNS_NAME}" --type=A --ttl=1 "${PACMAN_DST_PUBLIC_IP}"
+gcloud dns record-sets transaction execute -z=${ZONE_NAME}
+```
+
+### Remove Old Mongo Instance From Replica Set
+
+```bash
+MONGO_DST_POD=$(kubectl --context=gke_${GCP_PROJECT}_us-west1-b_gce-us-west1 get pod \
+    --selector="name=mongo" \
+    --output=jsonpath='{.items..metadata.name}')
+kubectl --context=gke_${GCP_PROJECT}_us-west1-b_gce-us-west1 \
+    exec -it ${MONGO_DST_POD} -- \
+    mongo --eval "rs.remove(\"${MONGO_SRC_PUBLIC_IP}:27017\")"
+```
+
+### Check Status to Verify Removal
+
+```bash
+kubectl --context=gke_${GCP_PROJECT}_us-west1-b_gce-us-west1 \
+    exec -it ${MONGO_DST_POD} -- \
+    mongo --eval "rs.status()"
+```
+
+### Remove Old Pac-man Cluster Resources
+
+```bash
+kubectl --context subdomain.example.com \
+    delete ns pacman
+```
+
+## Cleanup AWS Cluster
```

Comment:
I would put these cleanup instructions at the end because we still will talk about the automated way to migrate.